tests/e2e/performance/test_small_file_workload.py

```diff
@@ -424,64 +439,93 @@ def test_smallfile_workload(self, ripsaw, file_size, files, threads,
 
         Since the file_size is in Kb and the vol_size need to be in Gb, more
         calculation is needed.
+
         """
+        clients = sf_data['spec']['workload']['args']['clients']
+        test_files = files * threads * clients
+        dataset = test_files * file_size / constants.GB2KB
+        log.info(f'Total data set to use in the test is {dataset}')
         vol_size = int(files * threads * file_size * 3)
         vol_size = int(vol_size / constants.GB2KB)
         if vol_size < 100:
             vol_size = 100
-        sf_data['spec']['workload']['args']['storagesize'] = f"{vol_size}Gi"
+        sf_data['spec']['workload']['args']['storagesize'] = f'{vol_size}Gi'
 
         sf_obj = OCS(**sf_data)
         sf_obj.create()
-        log.info(f'The smallfile yaml file is {sf_data}')
 
         # wait for benchmark pods to get created - takes a while
-        for bench_pod in TimeoutSampler(
-            120, 3, get_pod_name_by_pattern, 'smallfile-client', 'my-ripsaw'
-        ):
-            try:
-                if bench_pod[0] is not None:
-                    small_file_client_pod = bench_pod[0]
-                    break
-            except IndexError:
-                log.info("Bench pod not ready yet")
+        clients = sf_data['spec']['workload']['args']['clients']
+        log.info(f'Going to run on {clients} pods, waiting for pods to start')
+        count = 10  # total number of retry for all benchmark pods to start
+        small_file_client_pods = {}
+        while count > 0:
```

Comment:
do we need the count, how about increasing the values in timeout sampler? will that help?

Reply:
changed the timeout, instead using the count.